testutils/networksuite/account.go

```diff
@@ -0,0 +1,85 @@
+package networksuite
+
+import (
+	"context"
+
+	"cosmossdk.io/math"
+	"github.com/cosmos/cosmos-sdk/client/tx"
+	secp256k1 "github.com/cosmos/cosmos-sdk/crypto/keys/secp256k1"
+	cryptotypes "github.com/cosmos/cosmos-sdk/crypto/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/types/tx/signing"
+	authsigning "github.com/cosmos/cosmos-sdk/x/auth/signing"
+)
+
+// Account is an abstraction around a cosmos-sdk private-key (account)
+type Account struct {
+	pk cryptotypes.PrivKey
+}
+
+// NewAccount returns a new account, with a randomly generated private-key.
+func NewAccount() *Account {
+	return &Account{
+		pk: secp256k1.GenPrivKey(),
+	}
+}
+
+// Address returns the address of the account.
+func (a Account) Address() sdk.AccAddress {
+	return sdk.AccAddress(a.pk.PubKey().Address())
+}
+
+// PubKey returns the public-key of the account.
+func (a Account) PubKey() cryptotypes.PubKey {
+	return a.pk.PubKey()
+}
+
+// CreateTx creates and signs a transaction, from the given messages
+func (a Account) CreateTx(ctx context.Context, accNum, seq, gasLimit, fee, timeoutHeight uint64, msgs ...sdk.Msg) ([]byte, error) {
```

Comment:
I pushed a similar function to chaintestutil https://github.com/skip-mev/chaintestutil/commit/7545b25770208ee7309ec55274416cc499419bf7

It seems like a function we can use across any package we want to test w/ that. Fine to leave this here and refactor it out later once we have a more stable set of chaintestutil capabilities implemented if you want to do that.

Reply:
I'll move to using chaintestutil

Reply:
On second thought, it looks like the helper there uses the val-keyring in the testsuite? I'd imagne that going forwards we want to be able to submit txs from accounts of our choice, and may want additional nuance in how we do things? Hence I'd prefer keeping thing like this, and refactoring out logic for tx-generation when we have a more educated opinion?

Reply:
Seems like we're going to move this to chaintestutil via https://github.com/skip-mev/chaintestutil/pull/5/files